packages/vm/src/vm.ts

```diff
@@ -223,13 +223,21 @@ export class VM extends AsyncEventEmitter<VMEvents> {
     // rather than deep copying the original ones since the copy of the `StateManager`
     // inside the EVM and EEI will be different than the `VM` level copy otherwise
     const eeiCopy = new EEI(stateCopy, commonCopy, blockchainCopy)
-    const evmCopy = new EVM({ eei: eeiCopy, common: commonCopy })
+    const evmCopy = new EVM({
+      eei: eeiCopy,
+      common: commonCopy,
+      allowUnlimitedContractSize: (this.evm as any)._allowUnlimitedContractSize,
+      customOpcodes: (this.evm as any)._customOpcodes,
+      customPrecompiles: (this.evm as any)._customPrecompiles,
+    })
```

Comment:
Hmm, this whole structure here is unlucky and not clean (it was before, nothing you introduced).

So one basic thing which also goes in here to note before: this issue was also about adopting the EVM `copy()` function, not only the VM one, so e.g. for these `customOpcodes` and `customPrecompiles` functions.

So: what is unlucky here that this is actually *not* using the (an) `EVM.copy()` function but doing a `new EVM(...)` instantiation.

The same goes for the `EEI`. The `EEI` has a copy function as well in the interface (`EEIInterface` in `types.ts` in the EVM package).

This is actually somewhat tricky due to what the comment above hints to (we want to have the same StateManager and Blockchain in the copied VM/EVM), but I think I got it now. 😋

So here we want to end up with the following:

1. Only do the EVM copy (so `EVM.copy()`), no other copies whatsoever (not state, blockchain, common, EEI).
2. For `VM.create()` use state, blockchain, common, EEI directly from the EVM with similar `any` casts as above

Can you please resubmit this PR and do things in the following order:

1. Adopt this `VM.copy()` function here as described (so using `EVM.copy()` + other adoptions), see if things are still working respectively tests are passing
2. Add the 3 missing parameters to `EVM.copy()` (allowUnlimitedContractSize, customOpcodes, customPrecompiles) (there should nothing of these needed to be made public, would rather be good if _allowUnlimitedContractSize can be made `protected` again)
3. Add new tests for EVM, see if things are working
4. Add the new options to `VM.copy()`, again, tests, see if things are working

Thanks! 🙂 

*first* adopt the `EVM.copy()` function with the